Reqnroll/StepArgumentTransformationAttribute.cs

```diff
@@ -17,14 +17,22 @@ public class StepArgumentTransformationAttribute : Attribute
         /// </summary>
         public string Name { get; set; }
 
-        public StepArgumentTransformationAttribute(string regex)
+        /// <summary>
+        /// Specifies the deterministic order for step argument transformations. Lower numbers have higher priority.
+        /// Before .NET 7, step argument transformations with the same priority will execute in a non-deterministic order.
```

Comment:
I don't understand the sentence about .NET 7. I think *in general* if you have multiple matching one without order or with the same order, then the behavior is non-deterministic. (Which is quite stupid IMHO, I think we should just throw an error, but that is a separate (breaking) change anyway.)

Reply:
Thanks for the comment! I've removed the sentence about .NET 7, but the original idea was referring to this statement from the docs:
> Starting with .NET 7, the ordering is deterministic based upon the metadata ordering in the assembly.

So in other words, the step argument transformations will be selected in the order they were originally declared in the source code when the assembly was compiled, which is quite deterministic (only for .NET 7+). I actually used it in one of the projects I'm working on to resolve the problem with the order before this change :)